File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/solution.java
class Solution {
  public int nearestValidPoint(int x, int y, int[][] points) {
    double minDistance = Double.POSITIVE_INFINITY;
    int result = -1;
    for (int index = 0; index < points.length; index++){
      int point[] = points[index];
      int currentx = point[0];
      int currenty = point[1];
      if (currentx == x || currenty == y){
        int distance = Math.abs(currentx - x) + Math.abs(currenty - y);
        if (distance < minDistance){
          minDistance = distance;
          result = index;
        }
      }
    }
    return result;
  }
}

File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/solution.js
/**
 * @param {number} x
 * @param {number} y
 * @param {number[][]} points
 * @return {number}
 */
 var nearestValidPoint = function(x, y, points) {
  let minDistance = Infinity;
  let result = -1;
  for (let index = 0; index < points.length; index++){
    let point = points[index];
    let currentx = point[0];
    let currenty = point[1];
    if (currentx === x || currenty === y){
      let distance = Math.abs(currentx - x) + Math.abs(currenty - y);
      if (distance < minDistance){
        minDistance = distance;
        result = index;
      }
    }
  }
  return result;
};